rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -253,3 +268,220 @@ def historical_uniswap_v2_lp_price(
+                    tick=entry[0][1],
+                    tick_bottom=entry[1][5],
+                    tick_top=entry[1][6],
+                    liquidity=entry[1][7],
+                    decimals=entry[2]['decimals'],
+                )
+                for entry in zip(slots_0, positions, tokens_b)
+            ]
+            for entry in zip(tokens_ids, pool_addresses, positions, price_ranges, tokens_a, tokens_b, amounts_a, amounts_b):  # noqa: 501
+                if entry[6] > ZERO or entry[7] > ZERO:
+                    entry[4].update({'amount': entry[6], 'address': entry[2][2]})
+                    entry[5].update({'amount': entry[7], 'address': entry[2][3]})
+                    balances.append(_decode_v3_result(userdb, entry, known_assets, unknown_assets))
+    return balances
+
+
+def compute_pool_address(
+    factory_address: ChecksumEthAddress,
+    token0_address: ChecksumEthAddress,
+    token1_address: ChecksumEthAddress,
+    fee: int,
+) -> ChecksumAddress:
+    """
+    Generate the pool address from the Uniswap Factory Address, pair of tokens
+    and the fee using CREATE2 opcode.
+    """
+    token_0 = to_checksum_address(token0_address)
+    token_1 = to_checksum_address(token1_address)
+    factory_address = to_checksum_address(factory_address)
+    parameters = []
+    # Sort the addresses
+    if int(token_0, 16) < int(token_1, 16):
+        parameters = [token_0, token_1, fee]
+    else:
+        parameters = [token_1, token_0, fee]
+    abi_encoded_1 = encode_abi(
+        ['address', 'address', 'uint24'],
+        parameters,
+    )
+    # pylint: disable=no-value-for-parameter
+    salt = Web3.solidityKeccak(abi_types=['bytes'], values=['0x' + abi_encoded_1.hex()])
+    abi_encoded_2 = encode_abi_packed(['address', 'bytes32'], (factory_address, salt))
+    # pylint: disable=no-value-for-parameter
+    address = Web3.solidityKeccak(abi_types=['bytes', 'bytes'],values=['0xff' + abi_encoded_2.hex(), POOL_INIT_CODE_HASH])[12:]  # noqa: 501
+    checksum_address = to_checksum_address(address.hex())
+    return checksum_address
+
+
+def calculate_price_range(
+    tick_lower: int,
+    tick_upper: int,
+    decimal_a: int,
+    decimal_b: int,
+
+) -> Tuple[FVal, FVal]:
+    """Calculates the price range for a Uniswap V3 LP position."""
+    sqrt_a = 1.0001**tick_lower
+    sqrt_b = 1.0001**tick_upper
+
+    sqrt_adjusted_a = sqrt_a * 10**(decimal_a - decimal_b)
+    sqrt_adjusted_b = sqrt_b * 10**(decimal_a - decimal_b)
+
+    return FVal(1 / sqrt_adjusted_b), FVal(1 / sqrt_adjusted_a)
+
+
+def calculate_amount_a(
+    tick_bottom: int,
+    liquidity: int,
+    tick_top: int,
+    decimals: int,
+    tick: int,
+) -> FVal:
+    """Calculates the amount of the first token in the Uniswap V3 LP position."""
+    sqrt_a = 1.0001**(tick_bottom / 2) * (2**96)
+    sqrt_b = 1.0001**(tick_top / 2) * (2**96)
+    sqrt = 1.0001**(tick / 2) * (2**96)
+    sqrt = max(min(sqrt, sqrt_b), sqrt_a)
```

Comment:
This code is repeated in both functions. Can't you combine them into one with one argument for calculating amount `a` and `b`? Also `amount_a` breaks the naming you were following... wouldn't it better `amount_0`?